zio-s3/src/main/scala/zio/s3/Live.scala

```diff
@@ -149,6 +151,29 @@ final class Live(unsafeClient: S3AsyncClient) extends S3 {
       }
       .unit
 
+  /**
+   * This method generates a `PresignedGetObjectRequest` containing an URL that allows the object to be downloaded
+   * without any credentials. Signature happens locally, it does not issue any network call. It's not a pure
+   * computation neither as it performs an IO by looking at the current time.
+   *
+   * See https://web.archive.org/web/20240621234636/https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html
+   * for further details regarding signature with S3.
+   */
+  override def presignGetObject(
```

Comment:
i dont think this code should be in the client. since there is nothing related to this. you could move this into the extra utils function

```
 def presignGetObject[R](
    bucketName: String,
    key: String,
    signatureDuration: Duration
  ): ZIO[S3Presigner, S3Exception, Unit] =   ZIO.serviceWithZIO[S3Presigner]{ s3Presigner =>

   //should wrap this unasync call with ZIO
   s3Presigner.presignGetObject(
        GetObjectPresignRequest
          .builder()
          .signatureDuration(signatureDuration)
          .getObjectRequest(GetObjectRequest.builder().bucket(bucketName).key(key).build())
          .build()
}


```

Reply:
Actually, the reason why we chose to put it in the client is that the configuration required is very similiar to the S3 client one.
The other solution is to provide another service with it's own layer definition, but then we should probably provide a layer for the configuration to avoid asking the same configuration twice to the user.
What do you think?